README.md

```diff
@@ -33,108 +34,242 @@ Menu
+
+
+### Metrics, tags and groups
+
+By default k6 measures and collects a lot of metrics about the things your scripts do - the duration of different script iterations, how much data was sent and received, how many HTTP requests were made, what was their duration, and even how long did the TLS handshake of a particual HTTPS request take. You see a summary of those built-in metrics in the output when you run the simplest possible k6 test, e.g. `k6 run github.com/loadimpact/k6/samples/http_get.js`. More information about the different built-in metrics collected by k6 (and how some of them can be accessed from inside of the scripts) can be seen in the docs [here](https://docs.k6.io/docs/result-metrics).
+
+k6 also allows the creation of user-defined `Counter`, `Gauge`, `Rate` and `Trend` metrics. They can be used to more precisely track and measure a custom subset of the things that k6 measures by default, or anything else the user wants, for example tracking non-timing information that is returned from the remote system. You can find more information about them [here](https://docs.k6.io/docs/result-metrics#section-custom-metrics) and a description of their APIs [here](https://docs.k6.io/docs/k6metrics).
+
+Every measurement metric in k6 comes with a set of key-value tags attached. Some of them are automatically added by k6 - for example a particular `http_req_duration` metric may have the `method=GET`, `status=200`, `url=https://loadimpact.com`, etc. system tags attached to it. Others can be added by users - globally for a test run via the `tags` [option](https://docs.k6.io/docs/options), or individually as a [parameter](https://docs.k6.io/docs/params-k6http) in a specific HTTP request, websocket connection, `userMetric.Add()` call, etc.
+
+These tags don't show in the simple summary at the end of a k6 test (unless you reference them in a [threshold](#checks-and-thresholds)), but they are invaluable for filtering and investigating k6 test results if you use any of the [outputs](#outputs) mentioned below. k6 also supports simple hierarchical groups for easier code and result organization. You can find more information about groups and system and user-defined tags [here](https://docs.k6.io/docs/tags-and-groups).
+
+### Checks and thresholds
+
+Checks and thresholds are some of the k6 features that make it very easy to use load tests like unit and functional tests and integrate them in a CI (continuous integration) workflow.
+
+[Checks](https://docs.k6.io/docs/checks) are like asserts, but differ in that they don't halt execution, instead they just store the result of the check, pass or fail, and let the script execution continue. Checks are great for codifying assertions relating to HTTP requests/responses, making sure the response code is 2xx for example.
```

Comment:
[Checks](https://docs.k6.io/docs/checks) are similar to asserts, but differ in that they don't halt execution. Instead they just store the result of the check, pass or fail, and let the script execution continue. Checks are great for codifying assertions relating to HTTP requests/responses. For example, making sure the response code is 2xx.